src/animation/play_once_animation.rb

```diff
@@ -0,0 +1,22 @@
+require 'gosu'
+require 'src/animation'
+
+class PlayOnceAnimation < Animation
+  def initialize(*args)
+    super(*args)
+    @played_frames = []
+  end
+
+  def start
+    frame = super
+    @played_frames << frame
+    frame
+  end
+
+  def stop
+    return start unless (frames - @played_frames).empty?
```

Comment:
I think that a simple count of played frame compared with the total frames could be more clear and simple than comparing list.